test/integration/podgc/podgc_test.go

```diff
@@ -255,9 +284,15 @@ func TestTerminatingOnOutOfServiceNode(t *testing.T) {
 				if err != nil {
 					t.Errorf("Error %q while waiting for the pod %q to be in expected phase", err, klog.KObj(pod))
 				}
-				_, cond := podutil.GetPodCondition(&pod.Status, v1.DisruptionTarget)
-				if cond != nil {
-					t.Errorf("Pod %q has an unexpected condition: %q", klog.KObj(pod), v1.DisruptionTarget)
+				numberCond, cond := podutil.GetPodCondition(&pod.Status, v1.DisruptionTarget)
```

Comment:
I don't think this change is needed. This part of the test verifies the DisruptionTarget condition isn't added in this scenario, even if PodDisruptionConditions is enabled. In this scenario the pod is being deleted not by PodGC (see "trigger termination of the pod, but with long grace period so that it is not removed immediate" in comment above). Because it is being deleted by another agent, it is responsibility of the other agent to add the condition, not PodGC.

Reply:
updated.